src/client/sandbox/event/focus-blur.js

```diff
@@ -64,85 +63,117 @@ export default class FocusBlurSandbox extends SandboxBase {
-                else
+                }
+                else if (browserUtils.isMSEdge && type === 'blur') {
+                    this.eventSimulator[focusInOutEventType](el);
+                    this.eventSimulator[type](el);
+                }
+                else {
                     this.eventSimulator[type](el);
+                    this.eventSimulator[focusInOutEventType](el);
+                }
             }
 
             callback();
         };
 
-        // NOTE: T239149 - TD15.1? - An error occurs during assertion creation on
-        // http://knockoutjs.com/examples/helloWorld.html in IE9.
-        if (browserUtils.isIE9 && this.shadowUI.getRoot() === el && (type === 'focus' || type === 'blur'))
+        if (this._isEventRaisedOnShadowUIRootInIE9(el))
             callback();
 
         if (el[type]) {
-            // NOTE: We should guarantee that activeElement will be changed, therefore we need to call the native
-            // focus/blur event. To guarantee that all focus/blur events are raised, we need to raise them manually.
-            var windowScroll                     = null;
-            var elementParents                   = [];
-            var nonScrollableParents             = [];
-            var nonScrollableParentsScrollValues = [];
-
-            if (preventScrolling)
-                windowScroll = styleUtils.getElementScroll(this.window);
-
-            if (browserUtils.isIE) {
-                elementParents = domUtils.getParents(el);
-
-                for (var i = 0; i < elementParents.length; i++) {
-                    if (styleUtils.get(elementParents[i], 'overflow') === 'hidden') {
-                        nonScrollableParents.push(elementParents[i]);
-                        nonScrollableParentsScrollValues.push(styleUtils.getElementScroll(elementParents[i]));
-                    }
-                }
-            }
-
-            var tempElement = null;
-
-            if (type === 'focus' && domUtils.isLabelElement(el) && el.htmlFor) {
-                tempElement = nativeMethods.getElementById.call(domUtils.findDocument(el), el.htmlFor);
-                if (tempElement)
-                    el = tempElement;
-                else {
-                    callback();
-                    return;
-                }
-            }
-
-            el[FocusBlurSandbox.getInternalEventFlag(type)] = true;
-
-            FocusBlurSandbox._getNativeMeth(el, type).call(el);
-
-            if (preventScrolling)
-                FocusBlurSandbox._restoreElementScroll(this.window, windowScroll);
-
-            if (browserUtils.isIE && nonScrollableParents.length) {
-                for (var j = 0, len = nonScrollableParents.length; j < len; j++)
-                    FocusBlurSandbox._restoreElementScroll(nonScrollableParents[j], nonScrollableParentsScrollValues[j]);
-            }
+            // NOTE: To guarantee that all focus/blur events are raised, we need to raise them manually.
```

Comment:
> This comment is not related with scrolling. 

This comment is placed on first line in brackets in which  we perform `_raiseEvent` calls.
https://github.com/miherlosev/testcafe-hammerhead/blob/74b2da15b2634991b3f66d762b90946d5d1e4ec6/src/client/sandbox/event/focus-blur.js#L175